Xamarin.Forms.Platform.Android/Renderers/RefreshViewRenderer.cs

```diff
@@ -0,0 +1,284 @@
+﻿using System;
+using System.ComponentModel;
+using Android.Content;
+using Android.OS;
+using Android.Support.V4.View;
+using Android.Support.V4.Widget;
+using Android.Support.V7.Widget;
+using Android.Views;
+using Android.Widget;
+using Xamarin.Forms.Internals;
+using AView = Android.Views.View;
+
+namespace Xamarin.Forms.Platform.Android
+{
+	public class RefreshViewRenderer : SwipeRefreshLayout, IVisualElementRenderer, IEffectControlProvider, SwipeRefreshLayout.IOnRefreshListener
+	{
+		bool _isDisposed;
+		bool _init;
+		bool _refreshing;
+		IVisualElementRenderer _renderer;
+		int? _defaultLabelFor;
+
+		public event EventHandler<VisualElementChangedEventArgs> ElementChanged;
+		public event EventHandler<PropertyChangedEventArgs> ElementPropertyChanged;
+
+		public RefreshViewRenderer(Context context)
+			: base(context)
+		{
+
+		}
+
+		public VisualElementTracker Tracker { get; private set; }
+
+		public ViewGroup ViewGroup => this;
+
+		public AView View => this;
+
+		public SwipeRefreshLayout SwipeRefreshLayout => View as SwipeRefreshLayout;
+
+		public VisualElement Element { get; private set; }
+
+		public RefreshView RefreshView => Element != null ? (RefreshView)Element : null;
+
+		public override bool Refreshing
+		{
+			get
+			{
+				return _refreshing;
+			}
+			set
+			{
+				_refreshing = value;
+
+				if (RefreshView != null && RefreshView.IsRefreshing != _refreshing)
+					RefreshView.IsRefreshing = _refreshing;
+
+				if (base.Refreshing == _refreshing)
+					return;
+
+				base.Refreshing = _refreshing;
+			}
+		}
+
+		public override bool CanChildScrollUp() => CanScrollUp(_renderer.View);
+
+		public void SetElement(VisualElement element)
+		{
+			var oldElement = Element;
+
+			if (oldElement != null)
+				oldElement.PropertyChanged -= HandlePropertyChanged;
+
+			Element = element;
+
+			if (Element != null)
+			{
+				UpdateContent();
+				Element.PropertyChanged += HandlePropertyChanged;
+			}
+
+			if (!_init)
+			{
+				_init = true;
+				Tracker = new VisualElementTracker(this);
+				SetOnRefreshListener(this);
+			}
+
+			UpdateColors();
+			UpdateIsRefreshing();
+			UpdateIsEnabled();
+
+			ElementChanged?.Invoke(this, new VisualElementChangedEventArgs(oldElement, Element));
+			EffectUtilities.RegisterEffectControlProvider(this, oldElement, element);
+		}
+
+		void UpdateContent()
+		{
+			if (RefreshView.Content == null)
+				return;
+
+			if (_renderer != null)
+			{
+				_renderer.View.RemoveFromParent();
+				_renderer.Dispose();
+				_renderer = null;
+			}
+
+			if (RefreshView.Content != null)
+			{
+				_renderer = Platform.CreateRenderer(RefreshView.Content, Context);
+
+				Platform.SetRenderer(RefreshView.Content, _renderer);
+
+				if (_renderer.View.Parent != null)
+					_renderer.View.RemoveFromParent();
+
+				using (var layout = new LayoutParams(LayoutParams.MatchParent, LayoutParams.MatchParent))
```

Comment:
`layoutParams` is probably a better name than `layout` as the latter implies an actual layout. Also, I'm not sure why we need a `using` here. Can't we pass the `LayoutParams` object without disposing it?